### ISSUE TITLE
Crash while running terraform plan

### ISSUE THREAD

**Issue by samuel** [hashicorp/terraform#8334]
This crash happens with Terraform v0.7.1 but not Terraform v0.7.0

We have a large terraform config using a number of modules we created so can't include that here, but some information about the crash:

crash log: https://gist.github.com/samuel/d3cd6c9fa4931d7847bf959fc7744160

This is the panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
2016/08/19 15:27:49 [DEBUG] plugin: terraform: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x51a350]
2016/08/19 15:27:49 [DEBUG] plugin: terraform:
2016/08/19 15:27:49 [DEBUG] plugin: terraform: goroutine 1508 [running]:
2016/08/19 15:27:49 [DEBUG] plugin: terraform: panic(0x23c4f80, 0xc4200100b0)
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/go/src/runtime/panic.go:500 +0x1a1
2016/08/19 15:27:49 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsVPCPeeringRead(0xc420b185a0, 0x2132c60, 0xc4202fe160, 0x0, 0x1a)
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_vpc_peering_connection.go:133 +0x560
2016/08/19 15:27:49 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc4209c0900, 0xc420727f40, 0x2132c60, 0xc4202fe160, 0xc420092230, 0x1, 0x18)
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:259 +0x131
2016/08/19 15:27:49 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc4209ba720, 0xc420727e00, 0xc420727f40, 0x0, 0xc420c88a98, 0x95ad85)
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:201 +0x91
2016/08/19 15:27:49 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc420878a80, 0xc420539b50, 0xc420539d80, 0x0, 0x0)
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:482 +0x4e
2016/08/19 15:27:49 [DEBUG] plugin: terraform: reflect.Value.call(0xc42032cc00, 0xc420450fb0, 0x13, 0x296ae45, 0x4, 0xc420d89ed0, 0x3, 0x3, 0x22092a0, 0xc420d657a0, ...)
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:434 +0x5c8
2016/08/19 15:27:49 [DEBUG] plugin: terraform: reflect.Value.Call(0xc42032cc00, 0xc420450fb0, 0x13, 0xc420d89ed0, 0x3, 0x3, 0xc421171c14, 0x100000000, 0xc4204bef01)
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:302 +0xa4
2016/08/19 15:27:49 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc4209d00c0, 0xc4209d0080, 0xc4209a6320, 0xc4201c9a00, 0xc420879200, 0x2134c60, 0xc420539b50, 0x16, 0x2134ca0, 0xc420539d80, ...)
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:383 +0x148
2016/08/19 15:27:49 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2016/08/19 15:27:49 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:477 +0x421
```

**Comment by samuel** [comment]
Seems to be caused by this commit: https://github.com/hashicorp/terraform/commit/5df0b08e86e80fb068269a401565d29fc1babaa0

building a binary from before that version works with no crash

**Comment by mitchellh** [comment]
Thanks for bisecting this for us! We'll take a look and get the regression fixed.

**Comment by kwilczynski** [comment]
@samuel apologies for the troubles.

I have resolved this in #8332.

**Comment by catsby** [comment]
Fixed in #8432 which I just merged, thanks!
